src/ripple/app/ledger/impl/LedgerConsensusImp.h

```diff
@@ -0,0 +1,354 @@
+#define RIPPLE_APP_LEDGER_IMPL_LEDGERCONSENSUSIMP_H_INCLUDED
+
+#include <BeastConfig.h>
+#include <ripple/app/ledger/impl/ConsensusImp.h>
+#include <ripple/app/ledger/impl/DisputedTx.h>
+#include <ripple/app/misc/CanonicalTXSet.h>
+#include <ripple/app/misc/FeeVote.h>
+#include <ripple/basics/CountedObject.h>
+#include <ripple/protocol/STValidation.h>
+#include <ripple/protocol/UintTypes.h>
+
+namespace ripple {
+
+/**
+  Provides the implementation for LedgerConsensus.
+
+  Achieves consensus on the next ledger.
+  This object is created when the consensus process starts, and
+  is destroyed when the process is complete.
+
+  Nearly everything herein is invoked with the master lock.
+
+  Two things need consensus:
+    1.  The set of transactions.
+    2.  The close time for the ledger.
+*/
+class LedgerConsensusImp
+    : public LedgerConsensus
+    , public std::enable_shared_from_this <LedgerConsensusImp>
+    , public CountedObject <LedgerConsensusImp>
+{
+private:
+    enum class State
+    {
+        // We haven't closed our ledger yet, but others might have
+        open,
+
+        // Establishing consensus
+        establish,
+
+        // We have closed on a transaction set
+        finished,
+
+        // We have accepted / validated a new last closed ledger
+        accepted,
+    };
+
+public:
+    /**
+     * The result of applying a transaction to a ledger.
+    */
+    enum {resultSuccess, resultFail, resultRetry};
+
+    static char const* getCountedObjectName () { return "LedgerConsensus"; }
+
+    LedgerConsensusImp(LedgerConsensusImp const&) = delete;
+    LedgerConsensusImp& operator=(LedgerConsensusImp const&) = delete;
+
+    ~LedgerConsensusImp () = default;
+
+    /**
+        @param previousProposers the number of participants in the last round
+        @param previousConvergeTime how long the last round took (ms)
+        @param inboundTransactions
+        @param localtx transactions issued by local clients
+        @param inboundTransactions the set of
+        @param localtx A set of local transactions to apply
+        @param prevLCLHash The hash of the Last Closed Ledger (LCL).
+        @param previousLedger Best guess of what the LCL was.
+        @param closeTime Closing time point of the LCL.
+        @param feeVote Our desired fee levels and voting logic.
+    */
+    LedgerConsensusImp (
+            ConsensusImp& consensus,
+            int previousProposers,
+            int previousConvergeTime,
+            InboundTransactions& inboundTransactions,
+            LocalTxs& localtx,
+            LedgerHash const & prevLCLHash,
+            Ledger::ref previousLedger,
+            std::uint32_t closeTime,
+            FeeVote& feeVote);
+
+    /**
+      Get the Json state of the consensus process.
+      Called by the consensus_info RPC.
+
+      @param full True if verbose response desired.
+      @return     The Json state.
+    */
+    Json::Value getJson (bool full);
+
+    uint256 getLCL ();
```

Comment:
There are a lot of nice comments around here.  Could we add one for this method so folks can know that `LCL` means `LastClosedLedger`?